test/spacerequest/spacerequest_assertions.go

```diff
@@ -69,6 +70,35 @@ func (a *Assertion) HasStatusTargetClusterURL(targetCluster string) *Assertion {
 	return a
 }
 
+func (a *Assertion) HasNamespaceAccess(namespaceAccess []toolchainv1alpha1.NamespaceAccess) *Assertion {
+	err := a.loadResource()
+	require.NoError(a.t, err)
+	assert.True(a.t, len(a.spaceRequest.Status.NamespaceAccess) > 0)
+
+	// check if each expected namespace has its access secret provisioned
+	for _, expectedNamespaceAccess := range namespaceAccess {
+		foundItem := toolchainv1alpha1.NamespaceAccess{}
+		for _, actualNamespaceAccess := range a.spaceRequest.Status.NamespaceAccess {
+			if actualNamespaceAccess.Name == expectedNamespaceAccess.Name {
+				foundItem = actualNamespaceAccess
+				break
+			}
+		}
+		assert.NotEmpty(a.t, foundItem, "unable to find namespace access", "namespace", expectedNamespaceAccess.Name)
+
+		// check that the secret was created
+		assert.NotEmpty(a.t, foundItem.SecretRef)
+		secret := &corev1.Secret{}
+		err = a.client.Get(context.TODO(), types.NamespacedName{
+			Namespace: a.spaceRequest.Namespace,
+			Name:      foundItem.SecretRef,
+		}, secret)
+		require.NoError(a.t, err)
+		assert.NotEmpty(a.t, secret)
```

Comment:
could we also verify the content of the secret?

Reply:
done in https://github.com/codeready-toolchain/host-operator/commit/03a04d98122bf138479a7714e44f3ea2db56f826